app/controllers/api/base_controller.rb

```diff
@@ -102,8 +102,10 @@ def resource_class_join(association, scope)
       resource_class.joins(association.name).merge(scope)
     end
 
-    def resource_scope_for_index(options = {})
-      resource_scope(options).search_for(*search_options).paginate(paginate_options)
+    def resource_scope_for_index(options = {}, filter = ->(default_scope) { default_scope })
+      scope = filter.call(resource_scope(options)).search_for(*search_options)
```

Comment:
I'm not sure this change is needed. scopes can be chained and the order of the chaining shouldn't matter, so for settings one you should be able to do `resource_scope_for_index.live_descendants`.